File: VB/Models/NorthwindDataProvider.vb
Imports System.Data
Imports System.Data.OleDb
Imports System.Web.Configuration

Public Class NorthwindDataProvider
	Public Shared Function GetProducts() As DataTable
		Dim connection As New OleDbConnection(WebConfigurationManager.ConnectionStrings("Northwind").ConnectionString)
		Dim selectCommand As New OleDbCommand("SELECT * FROM Products ORDER BY ProductID", connection)
		Dim da As New OleDbDataAdapter(selectCommand)
		Dim ds As New DataSet()

		selectCommand.Connection = connection
		da.Fill(ds, "Products")

		Return ds.Tables("Products")
	End Function
End Class

File: CS/Models/NorthwindDataProvider.cs
using System.Data;
using System.Data.OleDb;
using System.Web.Configuration;

public class NorthwindDataProvider {
    public static DataTable GetProducts() {
        OleDbConnection connection = new OleDbConnection(WebConfigurationManager.ConnectionStrings["Northwind"].ConnectionString);
        OleDbCommand selectCommand = new OleDbCommand("SELECT * FROM Products ORDER BY ProductID", connection);
        OleDbDataAdapter da = new OleDbDataAdapter(selectCommand);
        DataSet ds = new DataSet();

        selectCommand.Connection = connection;
        da.Fill(ds, "Products");

        return ds.Tables["Products"];
    }
}
